src/v/storage/log_manager.cc

```diff
@@ -312,6 +309,18 @@ ss::future<> log_manager::dispatch_topic_dir_deletion(ss::sstring dir) {
         });
     });
 }
+} // namespace
+ss::future<> log_manager::dispatch_topic_dir_deletion(ss::sstring dir) {
+    return delete_topic_dir_if_empty(dir).handle_exception_type(
+      [dir](const std::filesystem::filesystem_error& e) mutable {
+          vlog(
+            stlog.info,
+            "error deleting topic directory: {} - {}, retrying",
+            dir,
+            e);
+          return delete_topic_dir_if_empty(std::move(dir));
+      });
```

Comment:
this looks safe to me. but I had expected that you would still retry but it would be bounded with a back-off. do you think this is sufficient to address the issue? if so, lgtm.

Reply:
If `dispatch_topic_dir_deletion` is racing with the fiber that is removing the files that result in the target directory becoming empty, then wouldn't a single, immediate retry be just as flaky as a single shot removal with no retry?